models/article-model.js

```diff
@@ -18,23 +18,46 @@ exports.getArticleByIdQuery = ((id, next) => {
     });
 });
 
-exports.getArticlesQuery = ((next) => {
-    return db.query(
-        `SELECT a.article_id, a.title, a.topic, a.author, a.created_at, a.votes, a.article_img_url, count(c.article_id) as comment_count
-            FROM articles a
-            JOIN comments c
-                ON a.article_id = c.article_id
-            GROUP BY a.article_id
-            ORDER BY a.created_at DESC;`)
-    .then(({rows}) =>{
-        rows.map((row) => {
-            row.comment_count = Number(row.comment_count);
+exports.getArticlesQuery = ((next, topic='') => {
+    if (topic) {
+        // articles filtered by topic
+        return db.query(`SELECT a.article_id, a.title, a.topic, a.author, a.created_at, a.votes, a.article_img_url, count(c.article_id) as comment_count
+        FROM articles a
+        JOIN comments c
+            ON a.article_id = c.article_id
+        WHERE a.topic = $1
+        GROUP BY a.article_id
+        ORDER BY a.created_at DESC;`, [topic])
+
+        .then(({rows}) =>{
+            rows.map((row) => {
+                row.comment_count = Number(row.comment_count);
+            })
```

Comment:
This works! But ideally we'd get the comment_count as a number directly from the sql query try give the typecasting operator a go after your count(c.article_id)!   ::*datatype* can change the type of variables in pgsql. Here, we'd want an `int`.

Reply:
Already updated on a later PR - thanks :)